spack.yaml

```diff
@@ -3,40 +3,65 @@
 # It describes a set of packages to be installed, along with
 # configuration settings.
 spack:
-  definitions:
-  - compilers:
-    - intel@2021.10.0
-  - mpis:
-    - openmpi@4.1.5
-  - serial_packages:
-    - fortranxml@4.1.2
-  - packages:
-    - esmf@8.5.0+debug
-    - parallelio@2.6.2 build_type=RelWithDebInfo fflags='-qno-opt-dynamic-align -convert
-      big_endian -assume byterecl -ftz -traceback -assume realloc_lhs -fp-model source'
-      cflags='-qno-opt-dynamic-align -fp-model precise -std=gnu99'
-    - netcdf-c@4.9.2 build_system=cmake build_type=RelWithDebInfo
-    - netcdf-fortran@4.6.1
-    - fms@2023.02 precision=64 +large_file +deprecated_io ~gfs_phys ~openmp ~quad_precision
-      build_type=RelWithDebInfo
-    - access-om3 +install_libraries
   specs:
-  - matrix:
-    - [$mpis]
-    - [$%compilers]
-  - matrix:
-    - [$packages]
-    - [$^mpis]
-    - [$%compilers]
-  - matrix:
-    - [$serial_packages]
-    - [$%compilers]
+    - access-om3@git.2024.04.0
+  packages:
+    # Main Dependencies
+    access-om3-nuopc:
+      require:
+        - '@git.0.3.0'
+
+    # Other Dependencies
+    esmf:
+      require:
+        - '@8.5.0'
+        # - '-fp-model=precise'
+    parallelio:
+      require:
+        - '@2.6.2'
+    netcdf-c:
+      require:
+        - '@4.9.2'
+        - build_system=cmake build_type=RelWithDebInfo
+    netcdf-fortran:
+      require:
+        - '@4.6.1'
+    fms:
+      require:
+        - '@2023.02'
+    openmpi:
+      require:
+        - '@4.1.5'
+    fortranxml:
+      require:
+        - '@4.1.2'
+
+    all:
+      require:
+        - '%intel@2021.10.0'
+        - target=x86_64
   view: true
   concretizer:
     unify: true
   modules:
     default:
+      enable:
+        - tcl
       roots:
-        tcl: $spack/../modules/access-om3/0.x.0
+        tcl: $spack/../release/modules
+        lmod: $spack/../release/lmod
       tcl:
-        naming_scheme: '{name}/{version}'
+        hash_length: 0
+        include:
+          - access-om3
+        exclude_implicits: true
+        all:
+          autoload: direct
+          conflict:
+            - '{name}'
+          environment:
+            set:
+              'SPACK_{name}_ROOT': '{prefix}'
+        projections:
+          all: '{name}/{version}'
+          access-om3: '{name}/2024.04.0'
```

Comment:
Can we change this name to include the git hash of the commit used from COSIMA/access-om3 ? That will be easier to know what source code is being used ?

e.g. We want the module name to be :

`        - access-om3/63ed4a4777aeac0c62720ddbe7f85e4ed16f82eb_main`
instead of 
access-om3/2024.04.0

Reply:
Possible this is the wrong question. Ill need to try this again to understand how the bundle package is used vs the `access-om3-nuopc` package.

Reply:
`access-om3` is the version of overall deployment (including versions of `spack-packages`, `spack-config` etc) in one neat package. So it wouldn't make much sense to give the hash.
`access-om3-nuopc` (yes I know the name is hated and it can be changed) would make more sense to have the hash instead. I think you're able to do something like `access-om3-nuopc: '{name}/{hash}'` to get the version that you want. However, there is a check in the CI that will fail because it requires that the version specified in `spack.packages.*.requires:@VERSION` to be the same as the one in `spack.modules.default.tcl.projections.*:{name}/VERSION`. This means that you wouldn't be able to merge. 

@aidanheerdegen  should we loosen the version matching stuff to a warning instead of an error across the board? Or maybe make it configurable per-repo (more work)?

Reply:
Basically its needs to be easy for a researcher to identify what code they are running. Which can be aided by putting the hash in `config.yaml` of the payu config. I think well might need to `module load access-om3-nuopc/...` in payu config to access the `access-om3-...` binaries, so then the suggested change would work

Reply:
I might have to be bring down the TL ban-hammer and say no. 

Reasons:

1. There is CI that expects this format, so the module projects would be out of step with the releases and tagging on the repo
2. This would only ever be temporary. Public releases would not use this so there would be a point where the modules change how they're versioned. This would be confusing for users.
3. Hashes won't sort in order when users do `module list` so it is difficult to tell what is the most recent release
4. When OM3 has all model-components as spack packages their versions will be specified in the `spack.yaml` and `access-om3-nuopc` will cease to the source of truth for what sources are contained in a model version.